openwisp_controller/config/apps.py

```diff
@@ -135,6 +136,11 @@ def connect_signals(self):
             sender=self.config_model,
             dispatch_uid='devicegroup_templates_change_handler.backend_changed',
         )
+        pre_save.connect(
```

Comment:
I am thinking: would it work also with `post_save`? I would feel more comfortable with post_save, just in case for some reason the save operation fails we would avoid executing this.

Reply:
If we perform the operation in pre_save, then we won't be able to get the checksum of the template before saving. In short, we won't be able to determine if there's any significant change to the configuration.